python/pyspark/worker.py

```diff
@@ -146,7 +146,74 @@ def verify_result_type(result):
     )
 
 
-def wrap_cogrouped_map_pandas_udf(f, return_type, argspec):
+def verify_pandas_result(result, return_type, assign_cols_by_name):
+    import pandas as pd
+
+    if not isinstance(result, pd.DataFrame):
+        raise TypeError(
+            "Return type of the user-defined function should be "
+            "pandas.DataFrame, but is {}".format(type(result))
+        )
+
+    # check the schema of the result only if it is not empty or has columns
+    if not result.empty or len(result.columns) != 0:
+        # if any column name of the result is a string
+        # the column names of the result have to match the return type
+        #   see create_array in pyspark.sql.pandas.serializers.ArrowStreamPandasSerializer
+        field_names = set([field.name for field in return_type.fields])
+        column_names = set(result.columns)
+        if (
+            assign_cols_by_name
+            and any(isinstance(name, str) for name in result.columns)
+            and column_names != field_names
+        ):
+            limit = 5
```

Comment:
Hm, I wonder if this abbreviation is too much (and the character limit) ...

Reply:
e.g., would have to consider having a config to control this too ..

Reply:
> Hm, I wonder if this abbreviation is too much

Too much logic (better remove) or the limit too high?

Reply:
@HyukjinKwon I need your input here. Should I remove the abbreviation logic, or even printing the schema at all, to simplify things?

Reply:
@HyukjinKwon I have removed the schema and columns limit.